tests/backend/asgi/authenticated/test_api_version.py

```diff
@@ -0,0 +1,110 @@
+# Parsec Cloud (https://parsec.cloud) Copyright (c) AGPL-3.0 2016-present Scille SAS
+
+import pytest
+
+from parsec.api.version import API_V2_VERSION, API_VERSION, ApiVersion
+from parsec.core.types import LocalDevice
+from parsec._parsec import DateTime
+
+from tests.backend.asgi.authenticated.utils import (
+    dummy_request_with_reply,
+    send_authenticated_request,
+    signing_request,
+    build_authenticated_headers,
+)
+from tests.common.backend import RunningBackend
+
+
+@pytest.mark.trio
+async def test_api_version_missing_header(running_backend: RunningBackend, alice: LocalDevice):
+    client = running_backend.test_client()
+
+    org_id = alice.organization_id
+    dev_id = alice.device_id
+    timestamp = DateTime.now()
+
+    request, _reply_parser, _expected_reply = dummy_request_with_reply()
+
+    signature = signing_request(alice.signing_key, dev_id, timestamp, request)
+
+    headers = build_authenticated_headers(
+        api_version=None, author=dev_id, timestamp=timestamp, signature=signature
+    )
+
+    rep = await send_authenticated_request(client, org_id, headers, request)
```

Comment:
This code is too verbose and low level, instead you should do create a authenticated htttp client class with fixture to use it:

```python
class AuthenticatedHttpApiClient:
    def __init__(self, client: TestClientProtocol, device: LocalDevice):
        self.client = client
        self.device = device
        self.base_headers = {
            "Content-Type": "application/msgpack",
            "Api-Version": "PARSEC-SIGN-ED25519",
            [...]
            "Author": b64encode(device.device_id.encode("utf8")),
        }

    async def send_ping(self, ping, **kwargs):
        return await self.send({"cmd": "ping", "ping": ping}, ping_serializer, **kwargs)

    async def send(self, req, serializer, extra_headers: Dict[str, str]= {}, check_rep: bool = True):
        now = DateTime.now()
        body = serializer.req_dumps(req)
        headers = {
            **self.base_headers,
            "Timestamp": DateTime.now(),
            "Signature": signing_request(self.device.signing_key, self.device.device_id, now, body)
        }

        # Customize headers
        for k, v in extra_headers.items():
            if v is None:
                headers.pop(k, None)
            else:
                headers[k] = v

        rep = await self.client.post(f"/authenticated/{self.device.organization_id}", headers=headers, data=body)

        if check_rep:
            assert rep.status_code == 200
            rep_body = await rep.body()
            return serializer.rep_loads(rep_body)

        else:
            return rep

@pytest.fixture
def alice_http_client(alice, running_backend):
    client = running_backend.test_client()
    return AuthenticatedHttpApiClient(alice, client)


# and now you can do
# Test ok
rep = await alice_http_client.send_ping("foo")
assert rep == {"status": "ok", "pong": "foo"}

# Test error in transport
rep = await alice_http_client.send_ping("foo", check_rep=False, extra_headers={"timestamp": None})
assert rep.status_code == 400
```

Reply:
done